models/repo.go

```diff
@@ -593,15 +593,19 @@ func (repo *Repository) getReviewers(e Engine, doerID, posterID int64) ([]*User,
 	}
 
 	// This is a "public" repository:
-	// Any user that has write access or who is a watcher can be requested to review
+	// Any user that has write access, is a watcher or organization member can be requested to review
 	if err := e.
 		SQL("SELECT * FROM `user` WHERE id IN ( "+
-			"SELECT user_id FROM `access` WHERE repo_id = ? AND mode >= ? AND user_id NOT IN ( ?, ?) "+
+			"SELECT user_id FROM `access` WHERE repo_id = ? AND mode >= ? "+
 			"UNION "+
-			"SELECT user_id FROM `watch` WHERE repo_id = ? AND user_id NOT IN ( ?, ?) AND mode IN (?, ?) "+
-			") ORDER BY name",
-			repo.ID, AccessModeRead, doerID, posterID,
-			repo.ID, doerID, posterID, RepoWatchModeNormal, RepoWatchModeAuto).
+			"SELECT user_id FROM `watch` WHERE repo_id = ? AND mode IN (?, ?) "+
+			"UNION "+
+			"SELECT uid AS user_id FROM `org_user` WHERE org_id = ? "+
+			") AND id NOT IN (?, ?) ORDER BY name",
+			repo.ID, AccessModeRead,
```

Comment:
Comment says "any user with write access", but here we select for AccessModeRead? 🤔

Reply:
User with read access to code section can't leave review and shouldn't be selectable for review, this is common issue in many places in Gitea.

Reply:
GitHub also allows users with read permission to submit reviews on pull requests: https://docs.github.com/en/github/setting-up-and-managing-organizations-and-teams/repository-permission-levels-for-an-organization#repository-access-for-each-permission-level

Reply:
The issue here is that with Gitea you can personalize access to different sections. Team within org might not have access `Pull Requests` section (hence can't leave review) but is still selectable.
![firefox_2021-01-29_01-17-11](https://user-images.githubusercontent.com/1447794/106214741-ba7fe300-61cf-11eb-8c38-eda3cb29b228.png)

Reply:
But for an organization's **public** repositories you cannot remove read access to these sections via the team permissions this way. That only works for private repositories.

Reply:
Correct, this isn't blocker for this PR; just something I wanted to point out for the future since same "read only" logic is used for private repos.

Reply:
I agree @CirnoT, ONLY those members who have read access to pull requests should be listed as reviewers, otherwise when he will receive an email to review the PR and he click the link on the email, Gitea will give him a 404 because he has no permission to visit the URL.

Reply:
Yes, but this PR is not in conflict with that. For public repositories, everyone implicitly has read access to pull requests.